File: RenetikObjc/Library/Classes/Core/Categories/UIKit/UIApplication+CSExtension.m
//
// Created by Rene Dohan on 02/12/14.
// Copyright (c) 2014 creative_studio. All rights reserved.
//

#import "UIApplication+CSExtension.h"
#import "UIDevice+CSExtension.h"

@implementation UIApplication (CSExtension)

+ (UIWindow *)window {
    return [UIApplication.sharedApplication.windows lastObject];
}

+ (void)resignFirstResponder {
    [UIApplication.sharedApplication sendAction:@selector(resignFirstResponder) to:nil from:nil forEvent:nil];
}

- (UIApplication *)openNSURL:(NSURL *)url {
    if (UIDevice.isIOS10) [self openURL:url options:@{} completionHandler:nil];
    else [self openURL:url];
    return self;
}

- (UIApplication *)openURLString:(NSString *)url {
    [self openNSURL:[NSURL URLWithString:url]];
    return self;
}

+ (CGFloat)statusBarHeight {
    CGSize statusBarSize = [[UIApplication sharedApplication] statusBarFrame].size;
    return MIN(statusBarSize.width, statusBarSize.height);
}

@end

File: Sources/Renetik/extensions/UIKit/UIApplication+CSExtension.swift
//
// Created by Rene Dohan on 2/12/20.
//

import UIKit

public extension UIApplication {

    class var window: UIWindow? { UIApplication.shared.delegate?.window ?? nil }

    @objc class func resignFirstResponder() {
        UIApplication.shared.sendAction(#selector(resignFirstResponder), to: nil, from: nil, for: nil)
    }

    class func open(url: String) {
        open(url: URL(string: url)!)
    }

    class func open(url: URL) {
        UIApplication.shared.open(url, options: [:], completionHandler: nil)
    }

//    class var statusBarHeight: CGFloat {
//        let statusBarSize = UIApplication.shared.statusBarFrame.size
//        let height = min(statusBarSize.width, statusBarSize.height)
//        return height
//    }

//    class var statusBarBottom: CGFloat {
//        statusBarHeight
//    }
}